src/middlewared/middlewared/test/integration/utils/client.py

```diff
@@ -10,27 +10,39 @@
 from middlewared.client.utils import undefined
 
 __all__ = ["client", "host", "host_websocket_uri", "password", "session", "url", "websocket_url"]
+IS_HA = os.environ.get('SERVER_TYPE') == 'ENTERPRISE_HA'
 
 
 @contextlib.contextmanager
 def client(*, auth=undefined, auth_required=True, py_exceptions=True, log_py_exceptions=True, host_ip=None):
     if auth is undefined:
         auth = ("root", password())
 
+    uri = host_websocket_uri(host_ip)
     try:
-        with Client(host_websocket_uri(host_ip), py_exceptions=py_exceptions, log_py_exceptions=log_py_exceptions) as c:
+        with Client(uri, py_exceptions=py_exceptions, log_py_exceptions=log_py_exceptions) as c:
             if auth is not None:
                 logged_in = c.call("auth.login", *auth)
                 if auth_required:
                     assert logged_in
             yield c
-    except socket.timeout:
-        fail('socket timeout')
+    except socket.timeout as e:
+        if IS_HA:
+            # don't fail the entire run on HA systems since
+            # we perform failovers to and from each controller
+            # so timeouts could happen (which are expected)
+            raise e from None
+        else:
+            fail(f'socket timeout on URI: {uri!r} HOST_IP: {host_ip!r}')
 
 
 def host():
     if "NODE_A_IP" in os.environ:
+        # this is not to be confused with HA systems
+        # as it should only be set on NON-HA environments
```

Comment:
Why is it named `NODE_A` then?

Reply:
Don't know and don't care. Happened long before us and I'm not changing it right now.